Dockerfile

```diff
@@ -11,16 +11,44 @@ COPY src /usr/src/app/src
 RUN npm run build
 
 
+FROM golang:alpine AS gobuild
+
+RUN apk add make curl git && \
+	git clone https://github.com/prometheus/node_exporter && \
+	cd node_exporter && \
+	git checkout 5d3e2ce2ef927eeb44fc409d3cd1fbba884571f3 && \
+	make build && \
+	cp node_exporter /usr/bin/
+
+
 FROM base as main
 
-EXPOSE 80 443 3128
+EXPOSE 80 443 3128 9003
 
 RUN apt-get update -qq \
 	&& apt-get install -qy haproxy=1.8.* iptables --no-install-recommends \
 	&& apt-get clean \
 	&& rm -rf /var/lib/apt/lists/* /etc/apt/sources.list.d/*.list /etc/haproxy/* /etc/openvpn/* /etc/rsyslog.d/49-haproxy.conf \
 	&& ln -sf /usr/src/app/openvpn/scripts /etc/openvpn/scripts
 
+# MUSL-LIBC (for running 99%-static node_exporter binary built in gobuild stage)
+RUN wget http://www.musl-libc.org/releases/musl-1.1.10.tar.gz \
+	&& tar -xvf musl-1.1.10.tar.gz \
+	&& rm musl-1.1.10.tar.gz \
+	&& cd musl-1.1.10 \
+	&& ./configure \
+	&& make \
+	&& make install \
+	&& cd .. \
+	&& rm -rf musl-1.1.10
+
+# install nginx for auth-protecting node exporter metrics path
```

Comment:
haproxy is already configured to handle this, you can just add the node_exporter backend [here](https://github.com/balena-io/open-balena-vpn/blob/master/config/confd/templates/haproxy.cfg.tmpl#L36-L60).